AutoRest/Generators/Ruby/Azure.Ruby.Tests/RspecTests/azure_special_properties_spec.rb

```diff
@@ -153,7 +153,29 @@
     headers['x-ms-client-request-id'] = @validClientId
     result = @client.xms_client_request_id.get_async(headers).value!
     expect(result.response.status).to eq(200)
-    expect(result.request_id).to eq("123")
+    expect(result.response.headers['x-ms-request-id']).to eq("123")
+  end
+
+  # it 'should not overwrite x-ms-client-request-id' do

```

Comment:
Actually that is a valid test but looks like we have a bung that overrides client-request-id even though we set it to false. I'd prefer to keep it here as when we resolve that we can enable that test as well.  

---
In reply to: [69021567](https://github.com/Azure/autorest/pull/1188#discussion_r69021567) [](ancestors = 69021567)